packages/atlas/src/components/_inputs/AuctionDatePicker/AuctionDatePicker.tsx

```diff
@@ -1,93 +1,135 @@
 
-  const mappedItems = React.useMemo(
-    () =>
-      isPickDate && isValid(new Date(value))
-        ? [
-            ...items,
-            // selected date
-            {
-              value,
-              name: format(new Date(value), 'd MMM yyyy, HH:mm'),
-              hideInMenu: true,
-            },
+  const mappedItems: SelectItem<AuctionDatePickerValueWithPickDate>[] = useMemo(() => {
+    return isPickDate && pickedValue.type === 'date'
+      ? [
+          ...items,
+          {
+            value,
+            name: format(new Date(pickedValue.date), 'd MMM yyyy, HH:mm'),
+            hideInMenu: true,
             pickDateItem,
-          ]
-        : [...items, pickDateItem],
-    [isPickDate, items, pickDateItem, value]
-  )
+          },
+          pickDateItem,
+        ]
+      : [...items, pickDateItem]
+  }, [isPickDate, items, pickDateItem, pickedValue, value])
+
+  const handleSelect = (value?: AuctionDatePickerValueWithPickDate) => {
+    if (!value) {
+      return
+    }
+    if (value?.type !== 'pick-date') {
+      onChange(value)
+    }
+    setPickedValue(value)
+  }
+
+  const handlePickDate = (date: Date | null) => {
+    if (!date) {
+      return
+    }
+    setPickedValue({ type: 'date', date })
+    onChange({ type: 'date', date })
+    setStartDate(date)
+  }
 
   return (
     <Container>
-      <Select
-        size={size}
+      <Select<AuctionDatePickerValueWithPickDate>
+        size="small"
         label={label}
         labelTextProps={{ variant: 'h100', color: cVar('colorTextMuted'), secondary: true }}
         iconLeft={isPickDate ? <SvgControlsCalendar /> : undefined}
-        value={value}
+        value={pickedValue}
         items={mappedItems}
-        onChange={(value) => onChange(value ?? null)}
+        onChange={handleSelect}
         ref={selectRef}
         {...rest}
       />
-      <Popover offset={[0, 8]} ref={popOverRef} triggerMode="manual" triggerTarget={selectRef.current} trigger={null}>
+      <Popover
+        offset={[0, 8]}
+        ref={popOverRef}
+        triggerMode="manual"
+        triggerTarget={selectRef.current}
+        trigger={null}
+        onHide={() => {
+          if (pickedValue?.type === 'pick-date') {
+            setPickedValue({ durationDays: null, type: 'duration' })
+            onChange({ durationDays: null, type: 'duration' })
+          }
+        }}
+      >
         <DatePicker
           open
           inline
           selected={startDate}
-          onChange={(date) => {
-            if (date) {
-              setStartDate(date)
-              onChange?.(date)
-            }
-          }}
+          timeFormat="HH:mm"
+          onChange={handlePickDate}
           openToDate={minDate ?? undefined}
           minDate={minDate} // TODO: value to be discussed
-          maxDate={addMonths(new Date(), 5)} // TODO: value to be discussed
+          maxDate={maxDate || addMonths(new Date(), 5)} // TODO: value to be discussed
```

Comment:
I think both of those have to be defined by the consumer, not the component itself. Let's remove the `addMonths` and `TODO` comments and discuss that in `SetUp` usage